Tests/Microsoft.AppCenter.Test.Windows/AppCenterTest.cs

```diff
@@ -848,6 +846,88 @@ public void SetWrapperSdk()
             Assert.AreEqual(updatePackageHash, device.LiveUpdatePackageHash);
             Assert.AreEqual(runtimeVersion, device.WrapperRuntimeVersion);
         }
+
+        /// <summary>
+        /// Verify that AppCenter forwards SetMaxStorageSize to its channelGroup.
+        /// </summary>
+        [TestMethod]
+        public async Task SetMaxStorageSize()
+        {
+            // Set max storage size.
+            var dbSize = 2 * 1024 * 1024;
+            var storageTask = AppCenter.SetMaxStorageSizeAsync(dbSize);
+            AppCenter.Start("appsecret", typeof(MockAppCenterService));
+            await storageTask;
+
+            // Verify.
+            _channelGroupMock.Verify(channelGroup => channelGroup.SetMaxStorageSizeAsync(dbSize), Times.Once());
+        }
+
+        /// <summary>
+        /// Verify default value for max storage size is not set on AppCenter start.
+        /// </summary>
+        [TestMethod]
+        public void SetDefaultMaxStorageSizeOnStart()
+        {
+            // Set max storage default size.
+            AppCenter.Start("appsecret", typeof(MockAppCenterService));
+
+            // Verify.
+            _channelGroupMock.Verify(channelGroup => channelGroup.SetMaxStorageSizeAsync(10 * 1024 * 1024), Times.Once());
+        }
+
+        /// <summary>
+        /// Verify SetMaxStorageSize can only be called before the AppCenter start.
+        /// </summary>
+        [TestMethod]
+        public async Task CannotSetMaxStorageSizeAfterStart()
+        {
+            // Set max storage default size.
+            AppCenter.Start("appsecret", typeof(MockAppCenterService));
+            var dbSize = 2 * 1024 * 1024;
+
+            // Try to reset max storage size.
+            await AppCenter.SetMaxStorageSizeAsync(dbSize);
+
+            // Verify.
+            _channelGroupMock.Verify(channelGroup => channelGroup.SetMaxStorageSizeAsync(dbSize), Times.Never());
+        }
+
+        /// <summary>
+        /// Verify that SetMaxStorageSize can only be called once.
+        /// </summary>
+        [TestMethod]
+        public async Task CannotSetMaxStorageSizeMultipleTimes()
+        {
+            // Set max storage size first time before AppCenter start.
+            var dbSize = 2 * 1024 * 1024;
+            var taskFirst = AppCenter.SetMaxStorageSizeAsync(dbSize);
+
+            // Try to set max storage size second time before AppCenter start.
+            var taskSecond = AppCenter.SetMaxStorageSizeAsync(dbSize + 1);
+
+            // Start AppCenter and set storage max size.
+            AppCenter.Start("appsecret", typeof(MockAppCenterService));
+            await taskFirst;
+            await taskSecond;
+
+            // Verify.
+            _channelGroupMock.Verify(channelGroup => channelGroup.SetMaxStorageSizeAsync(dbSize), Times.Once());
```

Comment:
Maybe make sense to add this check after each `await`?

Reply:
We need to verify that _SetMaxStorageSizeAsync_ was called once after all tasks were completed.

Reply:
I guess it will be more clear if we add a check that it was called after the first `await` and a check that it wasn't called after the second `await`. Because in this case it is not clear in which of these `await` this API should be called.